src/game_api/aliases.hpp

```diff
@@ -190,3 +190,25 @@ enum class PAUSE_SCREEN : int64_t
     EXIT = 1 << 31,
 };
 ENUM_CLASS_FLAGS(PAUSE_SCREEN);
+
+enum class ENTITY_MASK
+{
+    PLAYER = 0x1,
+    MOUNT = 0x2,
+    MONSTER = 0x4,
+    ITEM = 0x8,
+    EXPLOSION = 0x10,
+    ROPE = 0x20,
+    FX = 0x40,
+    ACTIVEFLOOR = 0x80,
+    FLOOR = 0x100,
+    DECORATION = 0x200,
+    BG = 0x400,
+    SHADOW = 0x800,
+    LOGICAL = 0x1000,
+    WATER = 0x2000,
+    LAVA = 0x4000,
+    LIQUID = 0x6000,
+    NONE = 0x0,
```

Comment:
We use 0 as ANY, not NONE. This can be confusing, maybe just add both?

Reply:
Perhaps both, it was weird to do something like `(mask & ENTITY_MASK::WATER) == ENTITY_MASK::ANY`

Reply:
I would do `== 0`  in that case. Or the inverse `!= ENTITY_MASK::WATER`
Using mask to test for another mask feels weird
(and yes, i know it it's not allowed by default with enum class in C++)

Reply:
it's kinda annoying that it's not possible to just do `if (mask & ENTITY_MASK::WATER)` and you have to do either `(mask & ENTITY_MASK::WATER) == ENTITY_MASK::WATER` or `static_cast<uint32_t>(mask & ENTITY_MASK::WATER)`

Reply:
Would this be fine in `aliases.hpp`?
```cpp
template <class T>
bool test_flags(T flags, T test_flags)
{
    return (flags & test_flags) == test_flags;
}
```

Reply:
Whatever you think makes this enum easier to use and compare is fine.

Reply:
I also found another solution:
```cpp
namespace ENTITY_MASK
{
enum
{
    WATER = 0x2000,
    LAVA = 0x4000,
// etc.
}
}
```

Reply:
hmm yeah it works, and since what was being used before were raw integers, its still an improvement

Reply:
oh, it doesn't allow to use ENTITY_MASK as a type then

Reply:
I'm just gonna leave it like that, at least for now, as the rest of flags enums are

Reply:
> oh, it doesn't allow to use ENTITY_MASK as a type then

The `namespace` solution? it does, just have to give the enum itself name as well, it will look silly thou, with the type being `ENTITY_MASK::ENUM_NAME` and the enums only requiring the namespace.

Reply:
Also: I'm wondering if there is some compiler flag that would force the use of `ENUM_NAME::VAR` for the C style enums 🤔

Reply:
like this?
```cpp
namespace ENTITY_MASK
{
enum TYPE : uint32_t
{ ... }}
```
```cpp
uint8_t get_liquids_at(ENTITY_MASK::TYPE liquid_mask, float x, float y, LAYER layer)
```

Reply:
yeah, thou I'm not suggesting that this is good solution, just one that exist

Reply:
I think what I did now works well enough